benchmark-framework/src/main/java/io/openmessaging/benchmark/TestResult.java

```diff
@@ -45,6 +45,15 @@
     public List<Double> publishLatency9999pct = new ArrayList<>();
     public List<Double> publishLatencyMax = new ArrayList<>();
 
+    public List<Double> publishDelayLatencyAvg = new ArrayList<>();
```

Comment:
Does the publish delay only measures the delay we incur before calling `publish()` or the overall `delay + publish_latency` ?

Reply:
The former; I've decided to just provide the delay (or "wait time", in queuing theory parlee) as an additional metrics to just verify sanity of the load generation process., without fixing both `publish` and `endToEnd` latencies.

TBH I'm opened to suggestions here and just for reference other load generators eg https://github.com/apache/cassandra/commit/89f275c6587150907b90ed03ca515b12c0c6ae11 have decided to "patch" the so-called service-time (ie `publish` and `endToEnd` without coordination omission adjustment) adding `wait time` (ie `publish_delay`) to it, turning it into a response-time ( = wait time + service time) when the appropriate test type is defined (fixed rate). 

I've chose a more relaxed approach, in order to save existing results to not be valid anymore :)

Reply:
@merlimat any further feedbacks on this?
I hope to try it soon with Kafka (that has a blocking client iirc) to show how it can be useful

Reply:
If I understand this correctly this measure could be very helpful. For example, this measure could be used to help adjust the logic around finding the max production rate to be direct.

Reply:
> For example, this measure could be used to help adjust the logic around finding the max production rate to be direct.

Exactly or at least detect when there's something wrong that worth investigating:
- OS pause issues
- lack of CPU cycles because of AWS limitations
- client-side blocking behaviour(s)
- GC pauses
- ...